lib/aws-s3-creator-stack.ts

```diff
@@ -32,25 +31,28 @@ export class AwsS3CreatorStack extends cdk.Stack {
       enforceSSL: true,  // Ensure all requests to the S3 bucket use SSL
     });
 
-    // define an S3 bucket
-    const s3Bucket = new s3.Bucket(this, bucketName, {
-      bucketName: bucketName,
-      encryption: s3.BucketEncryption.S3_MANAGED,
-      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
-      publicReadAccess: false,
-      removalPolicy: removalPolicy,
-      autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
-      accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
-      versioned: true, // Enable versioning
-      serverAccessLogsBucket: logBucket,
-      enforceSSL: true,  // Ensure all requests to the S3 bucket use SSL
-    });
+    for (const s3BucketName of props.s3BucketNames) {
+      // define an S3 bucket
+      const s3Bucket = new s3.Bucket(this, `${props.resourcePrefix}-${s3BucketName}`, {
+        bucketName: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}`,
+        encryption: s3.BucketEncryption.S3_MANAGED,
+        blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+        publicReadAccess: false,
+        removalPolicy: removalPolicy,
+        autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
+        accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
+        versioned: true, // Enable versioning
+        serverAccessLogsBucket: logBucket,
+        enforceSSL: true,  // Ensure all requests to the S3 bucket use SSL
+        serverAccessLogsPrefix: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}-logs/`
+      });
 
-    // export s3Bucket name
-    new cdk.CfnOutput(this, `${props.appName}-${props.deployEnvironment}-s3BucketName`, {
-      value: s3Bucket.bucketName,
-      exportName: `${props.appName}-${props.deployEnvironment}-s3BucketName`,
-      description: 'The name of the S3 bucket.',
-    });
+      // export s3Bucket name
+      new cdk.CfnOutput(this, `${props.resourcePrefix}-${s3BucketName}-Export`, {
+        value: s3Bucket.bucketName,
+        exportName: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}-Export`,
+        description: 'The name of the S3 bucket.',
+      });
+    }
```

Comment:
**Suggestion:** Extract the bucket creation logic into a separate function to improve readability and maintainability.
> 🌉  **Medium Priority** 
 > [category: `maintainability`, importance: `**7**`]
> **Reason:** Extracting the bucket creation logic into a separate function as suggested would enhance code readability and maintainability, making it easier to manage and modify in the future.
 

```suggestion
    const createS3Bucket = (s3BucketName: string) => {
      const s3Bucket = new s3.Bucket(this, `${props.resourcePrefix}-${s3BucketName}`, {
        bucketName: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}`,
        encryption: s3.BucketEncryption.S3_MANAGED,
        blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
        publicReadAccess: false,
        removalPolicy: removalPolicy,
        autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
        accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
        versioned: true, // Enable versioning
        serverAccessLogsBucket: logBucket,
        enforceSSL: true,  // Ensure all requests to the S3 bucket use SSL
        serverAccessLogsPrefix: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}-logs/`
      });

      new cdk.CfnOutput(this, `${props.resourcePrefix}-${s3BucketName}-Export`, {
        value: s3Bucket.bucketName,
        exportName: `${props.resourcePrefix}-${props.deployRegion}-${s3BucketName}-Export`,
        description: 'The name of the S3 bucket.',
      });
    };

    for (const s3BucketName of props.s3BucketNames) {
      createS3Bucket(s3BucketName);
    }
```